src/backend/binaryen/wasm_type_gen.ts

```diff
@@ -752,6 +756,8 @@ export class WASMTypeGen {
         }
         if (type instanceof FunctionType) {
             return this.closureStructTypeMap.get(type) as binaryenCAPI.TypeRef;
+        } else if (type instanceof EnumType) {
+            return this.typeMap.get(type.memberType) as binaryenCAPI.TypeRef;
```

Comment:
Had better not to do special handling here, we can set the enum type as the key in typeMap.

Reply:
done